ca_on/people.py

```diff
@@ -22,12 +22,15 @@ def scrape(self):
             url = member.xpath('.//a//@href')[0]
             node = self.lxmlize(url, encoding='utf-8')
             fax = node.xpath('//div[@class="field field--name-field-fax-number field--type-string field--label-inline"]//div[@class="field__item"]//text()')
-            image = node.xpath('//div[@class="views-element-container block block-views block-views-blockmember-member-headshot"]//img/@src')[0]
+            if node.xpath('//div[@class="views-element-container block block-views block-views-blockmember-member-headshot"]//img/@src'):
+                image = node.xpath('//div[@class="views-element-container block block-views block-views-blockmember-member-headshot"]//img/@src')[0]
```

Comment:
To avoid repetition we typically do:

```
image = node.xpath('...')
if image:
    p.image = image[0]
```

This would have to appear after `p` is assigned, and replace the current `p.image = image` assignment.

Similarly for district and party.

That said, instead of adding these conditions, there should instead be a condition near the start of the for-loop that checks whether the page is for a vacant seat, and then `continue` if so.

Reply:
Thanks for this and the above feedback, I've refactored based on your comments